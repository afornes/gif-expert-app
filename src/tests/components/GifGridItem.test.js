import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Tests for GifGridItem', () => { 
  
  const title = 'Title';
  const url = 'https://localhost/foo.jpg';
  const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );
  
  test('should show a card', () => { 
    
    expect( wrapper ).toMatchSnapshot();
  });
  
  test('should have a paragraph with title', () => { 
    
    const p = wrapper.find('p');
    expect( p.text().trim() ).toBe( title );
  });
  
  test('should have an image with alt', () => { 
    
    const img = wrapper.find('img');
    expect(img.props().src).toBe(url);
    expect(img.props().alt).toBe(title);
  });
  
  test('should have animate__fadeIn', () => { 
    
    const div = wrapper.find('div');
    
    expect(div.props().className).toContain('animate__fadeIn');
  });
});