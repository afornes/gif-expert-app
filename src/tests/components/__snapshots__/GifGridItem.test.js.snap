// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifGridItem should show a card 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title"
    src="https://localhost/foo.jpg"
  />
  <p>
    Title
  </p>
</div>
`;
